templates/demo-store/.env

```diff
@@ -3,4 +3,4 @@
 SESSION_SECRET="foobar"
 PUBLIC_STOREFRONT_API_TOKEN="3b580e70970c4528da70c98e097c2fa0"
 PUBLIC_STOREFRONT_API_VERSION="2023-01"
-PUBLIC_STORE_DOMAIN="hydrogen-preview.myshopify.com"
+PUBLIC_STORE_DOMAIN="checkout.hydrogen.shop"
```

Comment:
Yeah, that's a good call out @frandiox -- curious to get your thoughts on my onboarding doc with a related notion here. 

One thing we could do as well -- and tell me if this wouldn't work -- if this is all about passing the right domain to analytics, is have the rest of the app rely on `shop.primaryDomain.host`

Reply:
Discussing this w/ Helen more — thinking about a separate option because if we do `myshopify` it also sets up the developer for failure.

```suggestion
// This token should be set to the public facing URL of your Liquid online store, learn more here: LINK_TO_DOCS
PUBLIC_STORE_DOMAIN="checkout.hydrogen.shop"
```

Reply:
I changed this back to `hydrogen-preview.myshopify.com`. As long as the checkout url returns the correct subdomain, it should be fine for now. We can do investigation if there are any performance impact using the `myshopify` path on the server side for a routing domain to Online Store.